docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
+     been configured for the `localhost` platform.
+   * Need to be able to update from the source location as part of reload.
 
-3. On start up, housekeep / archive logs and other items generated by the suite.
+3. On start up, archive any previous logs in a tar.gz.
+   * Propose to stop supporting this since, in most cases, we will use a
+     separate run directory each time a suite is installed.
+   * Continue to support a timestamped log dir for the special occasions where
+     we re-use an existing run directory (see the discussion in the CLI changes
+     below).
 
 4. On start up and reload, validate suite (using `--strict`).
    * Hopefully this can become a standard part of suite start-up rather than a
      separate step. This was difficult previously because cylc loads the suite
      after it has been detached?
-   * Consider making all validation strict? If we want really want to allow
-     naked dummy tasks in some circumstances then we would be better to add a
-     suite configuration for this.
+   * Consider making all validation strict?
+     * If we really want to allow naked dummy tasks in some circumstances then
+       we would be better to add a suite configuration for this?
+     * Note: we don't allow you to inherit from an undeclared ("naked") family
+       so why do we allow you to run an undeclared task?
 
-5. Handle `rose-suite.conf` functionality
-   * Pull items from other locations / sources on installation (e.g. svn).
-   * Configure Jinja2 / EmPy inputs.
-   * Set environment variables exported to the suite daemon?
-   * Continue to handle `rose-suite.conf` using a plugin preprocessor.
-   * Decide which features to support natively in Cylc and how.
+5. Handle
+   [rose-suite.conf](https://metomi.github.io/rose/doc/html/api/configuration/suite.html#rose:file:rose-suite.conf)
+   functionality including optional configurations. See
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practise but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
```

Comment:
For reference, here's a bit of context as to why this functionality has been removed:

We would expect that Rose would be installed in the same environment as Cylc so offering the option to configure the Rose version independently of the Cylc version does not make sense.

Note, however, that Rose will need to continue to use a wrapper script approach for locating remote installations as the Cylc environment is not activated in [remote and batch] task environments. We may decide to support this via a `ROSE_VERSION` environment variable as at present, though, since the Rose-Cylc couplings should be consistent across all installations, we could potentially rely on the `CYLC_VERSION` variable for the same functionality.

Perhaps one day version limiting could be done as part of https://github.com/cylc/cylc-flow/issues/3780?

Reply:
Not sure I fully understand this.
> the Cylc environment is not activated in [remote and batch] task environments

It will be activated in task job environments, to provide access to `cylc message` and other `cylc` commands, no?

Reply:
> It will be activated in task job environments, to provide access to cylc message and other cylc commands, no?

NO!!!!!!!

Cylc commands can be called via the wrapper script, activating the environment would change the Python interpreter and the entire Cylc stack.

At present the Cylc environment is only (accidentally) activated in local background/at jobs but that's going to change - https://github.com/cylc/cylc-flow/pull/3712

Reply:
Just thinking about this for a moment, I think we may require an additional `isodatetime` wrapper script if we wish to migrate users from the deprecated `rose date` and `cylc cycle-point` commands.

Reply:
> NO!!!!!!!
> 
> Cylc commands can be called via the wrapper script, activating the environment would change the Python interpreter and the entire Cylc stack

He heh, sorry, I wasn't thinking very deeply there - that is what I meant (the Cylc environment is activated in the job script, by the `cylc` wrapper, whenever a Cylc command is called) - but yeah that doesn't help separate use of the Rose CLI.

Reply:
> Just thinking about this for a moment, I think we may require an additional `isodatetime` wrapper script if we wish to migrate users from the deprecated `rose date` and `cylc cycle-point` commands.

I fear you're right - yuck!

Reply:
(Commented on the Cylc wrapper script PR suggesting using `CYLC_VERSION` for the `isodatetime` wrapper script - https://github.com/cylc/cylc-flow/pull/3754#issuecomment-709031881)